pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -451,7 +451,16 @@ func (f *ring0Factory) Resumer(info *resource.Info) ([]byte, error) {
 }
 
 func (f *ring0Factory) DefaultNamespace() (string, bool, error) {
-	return f.clientConfig.Namespace()
+	namespace, boolRetval, err := f.clientConfig.Namespace()
```

Comment:
hrmm... here be dragons... leaving an invalid value being returned from clientConfig.Namespace() is no good. for example, see this logic that checks whether the namespace was passed and should override:
```
func (config *DirectClientConfig) Namespace() (string, bool, error) {
	if config.overrides != nil && config.overrides.Context.Namespace != "" {
		// In the event we have an empty config but we do have a namespace override, we should return
		// the namespace override instead of having config.ConfirmUsable() return an error. This allows
		// things like in-cluster clients to execute `kubectl get pods --namespace=foo` and have the
		// --namespace flag honored instead of being ignored.
		return config.overrides.Context.Namespace, true, nil
	}
```

this has to go even deeper, probably into the Namespace() method, and/or possibly into the getContext() method. It's also an open question whether we should accept the `namespace/...` syntax everywhere (e.g. in kubeconfig files, etc) or just via the command-line `-n` flag